dagmc/dagnav.py

```diff
@@ -279,6 +279,11 @@ def get_triangle_coordinate_mapping(self, compress=False):
         tri_map = {eh: c for eh, c in zip(triangle_handles, conn)}
         return tri_map, coords
 
+    def delete(self):
+        """Delete this group from the DAGMC file."""
+        self.model.mb.delete_entity(self.handle)
```

Comment:
Should we delete the entry in `model.groups`?

Reply:
Ah, sorry for the quick-merge! I think we're okay here though as the `dict` returned by the `model.groups` property is generated on-the-fly, so the next time it's accessed the groups should reflect those currently in the model.

Previous `dict`'s generated by this property will become stale after a group is deleted, but that will have to be up to the user to track I'm afraid.